src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx

```diff
@@ -77,101 +101,380 @@ export const Confirm: React.FC<ConfirmProps> = ({ accountId, onNext }) => {
+    opportunityData?.stakedAmountCryptoBaseUnit,
+    state?.withdraw.cryptoAmount,
+  ])
+
+  const getPreWithdrawInput: () => Promise<SendInput | undefined> = useCallback(async () => {
+    if (
+      !(
+        accountId &&
+        assetId &&
+        state?.withdraw?.estimatedGasCrypto &&
+        opportunityData?.stakedAmountCryptoBaseUnit
+      )
+    )
+      return
+
+    try {
+      const estimatedFees = await estimateFees(await getEstimateFeesArgs())
+      const amountCryptoBaseUnit = bnOrZero(state?.withdraw.cryptoAmount).times(
+        bn(10).pow(asset.precision),
+      )
+
+      const bps = getWithdrawBps({
+        withdrawAmountCryptoBaseUnit: amountCryptoBaseUnit,
+        stakedAmountCryptoBaseUnit: opportunityData?.stakedAmountCryptoBaseUnit,
+        rewardsamountCryptoBaseUnit: opportunityData?.rewardsAmountsCryptoBaseUnit?.[0] ?? '0',
+      })
+
+      const quote = await getThorchainSaversWithdrawQuote({ asset, accountId, bps })
+      const accountAddress = await getMaybeUtxoAccountAddress()
+
+      const sendInput: SendInput = {
+        cryptoAmount: '',
+        asset,
+        from: '', // Let coinselect do its magic here
+        to: accountAddress,
+        sendMax: true,
+        accountId,
+        amountFieldError: '',
+        cryptoSymbol: asset.symbol,
+        estimatedFees,
+        feeType: FeeDataKey.Fast,
+        fiatAmount: '',
+        fiatSymbol: selectedCurrency,
+        vanityAddress: '',
+        input: quote.inbound_address,
+      }
+
+      return sendInput
+    } catch (e) {
+      moduleLogger.error({ fn: 'getDepositInput', e }, 'Error building THORChain savers Tx')
+    }
+  }, [
+    accountId,
+    assetId,
+    state?.withdraw?.estimatedGasCrypto,
+    state?.withdraw.cryptoAmount,
+    opportunityData?.stakedAmountCryptoBaseUnit,
+    opportunityData?.rewardsAmountsCryptoBaseUnit,
+    getEstimateFeesArgs,
+    asset,
+    getMaybeUtxoAccountAddress,
+    selectedCurrency,
+  ])
+
+  const getWithdrawInput: () => Promise<SendInput | undefined> = useCallback(async () => {
+    if (!(accountId && assetId && opportunityData?.stakedAmountCryptoBaseUnit)) return
+
+    try {
+      const estimatedFees = await estimateFees(await getEstimateFeesArgs())
+      const amountCryptoBaseUnit = bnOrZero(state?.withdraw.cryptoAmount).times(
+        bn(10).pow(asset.precision),
+      )
+
+      const withdrawBps = getWithdrawBps({
+        withdrawAmountCryptoBaseUnit: amountCryptoBaseUnit,
+        stakedAmountCryptoBaseUnit: opportunityData?.stakedAmountCryptoBaseUnit,
+        rewardsamountCryptoBaseUnit: opportunityData?.rewardsAmountsCryptoBaseUnit?.[0] ?? '0',
+      })
+
+      const quote = await getThorchainSaversWithdrawQuote({ asset, accountId, bps: withdrawBps })
+
+      if (!quote) throw new Error('Error getting THORChain savers withdraw quote')
+
+      const { dust_amount } = quote
+
+      const accountAddress = await getMaybeUtxoAccountAddress()
```

Comment:
guard against this being empty

Reply:
d172d88a6cd573fcd24836e27b2ec2a35501ff19